app/services/importer/scraper/workday_cookie_fetcher.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Importer
+  module Scraper
+    class WorkdayCookieFetcher < ApplicationTask
+      include Capybara::DSL
+
+      def initialize(details)
+        @create_account_url = details[:create_account_url]
+        @sign_in_url = details[:sign_in_url]
+        @target = details[:target_url]
+        Capybara.configure do |config|
+          config.test_id = :'data-automation-id'
+          config.default_max_wait_time = 5
+        end
+        @session = Capybara::Session.new(:selenium) # replace with :selenium_chrome_headless
+      end
+
+      def call
+        return nil unless processable
+
+        using_session(@session) do
+          process
+        end
+      end
+
+      private
+
+      def processable
+        @sign_in_url && @create_account_url
+      end
+
+      def process
+        sign_in
+        create_account unless login_success?
+        return_cookie
+      ensure
+        @session.quit
+      end
+
+      def cookie_has_updated?
+        old_cookie = @cookie
+        log_cookie
+        @cookie != old_cookie
+      end
+
+      def create_account
+        click_button('createAccountLink')
+        log_cookie
+        fill_in('email', with: ENV.fetch('WORKDAY_EMAIL'))
```

Comment:
Do we have a fixed login at the moment? This should probably be generated / rotated somehow to avoid them suspecting abuse

Reply:
Great solution btw as a first setup

Reply:
I'll think about this some more. If we rotate or generate, we'll need to store the username/password/company_endpoint association somewhere to enable signing in again in future. Or maybe not, maybe we dispense with logins and just create a new throwaway account each time? (Would be annoying as hell for Workday, but would prob simplify/save time on our end). Let's discuss at some point.